src/main/java/gui/logic/BusinessLogic.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class BusinessLogic {
+
+    public static final String baseIRI = "http://webprotege.stanford.edu/";
+
+    public static MyFactory getFactory() {
+        OWLOntologyManager ontologyManager = OWLManager.createOWLOntologyManager();
+        OWLOntology ontology = null;
+        try {
+            ontology = ontologyManager.loadOntologyFromOntologyDocument(new File("src/main/resources/traffic_ontology.owl"));
+        } catch (OWLOntologyCreationException e) {
+            System.err.println("Problem during loading ontology");
+            e.printStackTrace();
+        }
+        return new MyFactory(ontology);
+    }
+
+    // na razie na sztywno korzystamy z BaseScenarioGenerator2
+    public static Model getModelFromGenerator(MyFactory factory) {
+        BaseScenarioGenerator2 generator = new BaseScenarioGenerator2(factory, baseIRI);
+        Model model = null;
+        try {
+            model = generator.generate();
+        } catch (NoSuchMethodException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        } catch (InvocationTargetException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        }
+        DecisionGenerator decisionGenerator = new DecisionGenerator(factory, baseIRI);
+        decisionGenerator.generate(model);
+        return model;
+    }
+
+    public static Map<Decision, Set<Actor>> getCollidedEntities(IConsequenceContainer consequenceContainer, MyFactory factory, Model scenarioModel) {
+        CollisionConsequencePredictor collisionConsequencePredictor =
+                new CollisionConsequencePredictor(consequenceContainer, factory, scenarioModel);
+        SimulatorEngine simulatorEngine = new SimulatorEngine(scenarioModel, collisionConsequencePredictor);
+        Set leftLanes = scenarioModel.getLanes().get(Model.Side.LEFT).entrySet();
+        int lastLeftLane = leftLanes.size();
+        Set rightLanes = scenarioModel.getLanes().get(Model.Side.RIGHT).entrySet();
+        int lastRightLane = rightLanes.size();
+        Map<Decision, Set<Actor>> collidedEntities = simulatorEngine.simulateAll(lastLeftLane, lastRightLane);
+
+        // pytanie czy może to być w tym miejscu, bo umieszczenie tego później miałoby sens,
+        // jeśli korzytalibyśmy z MoralDilemmaDetector
+        try {
+            factory.saveOwlOntology();
+        } catch (OWLOntologyStorageException e) {
+            System.err.println("Problem during saving ontology");
+            e.printStackTrace();
+        }
+        return collidedEntities;
+    }
+
+    // optymalnie jechać cały nie wykonywać manewrów,
```

Comment:
Funkcja stworzona po to, żeby podać podać jaka decyzja jest najlepsza. Ogólnie, jakby jedna miała najniższą sumę to sprawa jest oczywista, ale tutaj u nas na razie często zdarza się kilka decyzji po 0 punktów.

I tutaj dałem na razie taką logikę, że jak chcemy dostać tylko jedną najlepszą decyzję to wybieramy w kolejności follow, zmiany pasa na prawo, później na lewo, skręt w prawo, wszystko inne.

Powodem tej funkcji jest, żeby zawsze zwracało nam tą samą najlepszą decyzję, a nie w zależności od tego jak przeiterujemy po hashmapie.

Ale też ogólnie do dyskusji, może być na żywo, jak widzicie tą strategię i czy w ogóle chcemy wybierać jedną najlepszą decyzję. Bo ogólnie to może właśnie z racji ruchu prawo stronnego powinniśmy zmieniać pas na lewo. Ja w tej strategii, kierowałem się tym, że lepiej jest zmienić pas na prawo, żeby nie wyjechać na pas z ruchem w przeciwnym kierunku. Ale to tutaj zależy czy mamy drogę jednokierunkową czy dwukierunkową.

Reply:
Jak dla mnie narazie jest ok tak jak zrobiłeś. To w sumie zależy. Tak jak mówisz, rodzaj drogi, kierunek ruchu, nawet pasy ciągłe czy przerywane (łamanie prawa). Jak jest kilka decyzji o koszcie 0 to może być tak jak zrobiłeś narazie, z tym, że fajnie by było może jakoś zhighlightować, że jest więcej spoko decyzji. (podświetlić/napisać pod spodem czy coś xD)